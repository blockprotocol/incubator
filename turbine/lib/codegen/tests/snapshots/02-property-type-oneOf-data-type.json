[
  {
    "$schema": "https://blockprotocol.org/types/modules/graph/0.3/schema/property-type",
    "kind": "propertyType",
    "$id": "https://blockprotocol.org/@alice/property-type/user-id/v/1",
    "title": "User ID",
    "oneOf": [
      {
        "$ref": "https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1"
      },
      {
        "$ref": "https://blockprotocol.org/@blockprotocol/types/data-type/number/v/1"
      }
    ]
  }
]
