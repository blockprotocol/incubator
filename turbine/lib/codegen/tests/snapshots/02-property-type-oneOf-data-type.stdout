https_blockprotocol_org_alice_property_type_user_id.rs

pub enum UserId {
    Variant0(Text),
    Variant1(Number),
}
pub enum UserIdRef<'a> {
    Variant0(Text::Ref<'a>),
    Variant1(Number::Ref<'a>),
}
pub enum UserIdMut<'a> {
    Variant0(Text::Mut<'a>),
    Variant1(Number::Mut<'a>),
}
